File: 240514/수열 조작/sequence-manipulation.java
import java.util.*;

public class Main {
    static Scanner sc = new Scanner(System.in);
    static int n = sc.nextInt();
    static ArrayDeque<Integer> deque = new ArrayDeque<Integer>();

    public static void main(String[] args) {
        for(int i = 1; i <= n; i++) {
            deque.addLast(i);
        }
        operation();
        System.out.println(deque.pollFirst());
    }

    static void operation(){
        while(deque.size() > 1) {
            deque.pollFirst();
            deque.addLast(deque.pollFirst());
        }
    }

}

File: 240514/수열 조작/sequence-manipulation.kt
import java.util.*

val sc = Scanner(System.`in`)
val n = sc.nextInt()
val deque: ArrayDeque<Int> = ArrayDeque()

fun main() {
    for(i in 1 .. n) {
        deque.addLast(i)
    }
    operate()
    println(deque.pollFirst())
}

fun operate() {
    while(deque.size > 1) {
        deque.pollFirst()
        if(deque.size > 1) {
            deque.addLast(deque.pollFirst())
        }else {
            return
        }
    }
}